File: 01-math/03-check-a-num-is-palindrome/03-check-a-num-is-palindrome.rb
def check_a_num_is_palindrome(num)
  temp, reversed = num, 0
  while temp > 0
    reversed *= 10
    reversed += temp % 10
    temp /= 10
  end
  reversed == num
end

puts check_a_num_is_palindrome 9999
puts check_a_num_is_palindrome 9998

File: 01-math/03-check-a-num-is-palindrome/03-check-a-num-is-palindrome.ts
const check_a_num_is_palindrome = (num: number) => {
  let reversed = 0
  let temp = num
  while (temp > 0) {
    reversed *= 10
    reversed += temp % 10
    temp = Math.floor(temp / 10)
  }
  return reversed === num
}

console.log(check_a_num_is_palindrome(9999))
console.log(check_a_num_is_palindrome(9998))
